src/App.js

```diff
@@ -1,25 +1,29 @@
-import logo from './logo.svg';
-import './App.css';
+import { createContext, useState } from "react";
+import { Route, Routes } from "react-router-dom";
+import "./App.css";
+import Articles from "./components/Articles";
+import Description from "./components/Description";
+import Header from "./components/Header";
+import Nav from "./components/Nav";
+
+const articleContext = createContext();
 
 function App() {
+  const [articles, setArticles] = useState([]);
   return (
+      <articleContext.Provider value={{ articles, setArticles }}>
```

Comment:
React Context is a very useful tool to share state in a component tree but i don't think it's appropriate in this case

At the moment when `setArticles` is called your *whole app* will re-render

You should try to hold your state as low down as possible in your component tree.

Reply:
Actually, according to my components diagram, I need to access the article's state on different levels of component so I thought of using context in the beginning, should I access it through props for now?